scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -746,7 +762,23 @@ class FossId internal constructor(
             "${pendingFiles.size} pending files have been returned for scan '$scanCode'."
         }
 
-        return RawResults(identifiedFiles, markedAsIdentifiedFiles, listIgnoredFiles, pendingFiles)
+        val snippets = runBlocking(Dispatchers.IO) {
+            pendingFiles.map {
+                async {
+                    logger.info { "Listing snippet for $it..." }
+                    val snippetResponse = service.listSnippets(config.user, config.apiKey, scanCode, it)
+                        .checkResponse("list snippets")
+                    val snippets = requireNotNull(snippetResponse.data) {
```

Comment:
`require(NotNull)` should be used to verify arguments / user input. Use `check(NotNull)` to verify state.